lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -3387,9 +3395,19 @@ bool PartitionCloner::finalizeOriginal() {
   // Next, add sends back of any values that are used by the host code, and
   // remove the original instruction.
   for (auto inst : instructionsToRemove) {
-    // These insts cannot contain types like unconditional branch, which do not
-    // have getResults() defined.
-    assert(!isa<NonValueInstruction>(inst));
+    if (isUserIgnoredByPartitioning(inst)) {
+      // TODO: If we are removing retain/release insts over a tensor value `a`,
+      // and `a` has uses in the host code after the tensor program region, we
+      // need to rebalance its retain/release count.
+      inst->eraseFromParent();
+      continue;
+    }
+    // These insts cannot contain types like unconditional branch, which do
+    // not have getResults() defined.
+    if (isa<NonValueInstruction>(inst)) {
```

Comment:
Please turn this into an assert.  I know you lose the dump() but this shouldn't be triggering enough to warrant it.